components/CorrectCounter/CorrectCounter.js

```diff
@@ -0,0 +1,78 @@
+import styled from "styled-components";
+import RoundButton from "../Buttons/RoundButton";
+import MarkAsIncorrect from "@/public/icons/MarkAsIncorrect.svg";
+import MarkAsCorrect from "@/public/icons/MarkAsCorrect.svg";
+import Stack from "@/public/icons/Stack.svg";
+import { useRouter } from "next/router";
+
+export default function CorrectCounter({
+  variant,
+  actionMode,
+  getIncorrectFlashcardsFromCollection,
+  getCorrectFlashcardsFromCollection,
+  id,
+  isFooter,
+}) {
+  const router = useRouter();
+
+  if (variant !== "correct" && variant !== "incorrect" && variant !== "all") {
+    return console.log("This variant doesn't exist.");
+  }
+
+  const numberOfIncorrectFlashcards =
+    getIncorrectFlashcardsFromCollection(id).length;
+  const numberOfCorrectFlashcards =
+    getCorrectFlashcardsFromCollection(id).length;
+  const numberOfAllFlashcards =
+    numberOfCorrectFlashcards + numberOfIncorrectFlashcards;
+
+  const correctVariants = {
+    incorrect: {
+      icon: <MarkAsIncorrect />,
+      count: numberOfIncorrectFlashcards,
+    },
+    correct: {
+      icon: <MarkAsCorrect />,
+      count: numberOfCorrectFlashcards,
+    },
+    all: {
+      icon: <Stack />,
+      count: numberOfAllFlashcards,
+    },
+  };
+
+  const { icon, count } = correctVariants[variant];
+
+  function handleRedirect() {
+    if (variant === "correct") {
+      router.push(`/${id}/learned`);
+    } else if (variant === "incorrect") {
+      router.push(`/${id}/to-learn`);
+    } else if (variant === "all") {
+      router.push(`/${id}`);
+    }
+  }
+
+  return (
+    <CorrectCounterWrapper onClick={handleRedirect}>
+      <RoundButton
+        isFooter={isFooter}
+        content={icon}
+        type="button"
+        variant={variant}
+        actionMode={actionMode}
+      />
+      <Counter>{count}</Counter>
+    </CorrectCounterWrapper>
+  );
+}
+
+const CorrectCounterWrapper = styled.article`
```

Comment:
You are currently using this element in your navigation as a clickable element to navigate between pages.

Ideally you want to work with the `Link` component in Next.js to navigate between pages.

If you need to compute the path and redirect users imperatively, which you apparently need to do, I suggest you work with a button element instead.

Reply:
Well, I don't *have* to calculate the path and redirect users that way. I decided to use the variant notation so that I don't have to specify as many props. In each variant there is a different icon, a different count and a different link. And since everything is always the same in a variant, I thought it was absurd to specify all three things separately. That's why I solved it this way. Don't you think that's good? In any case, I'll implement your suggestion to use a button element rather than an article directly, thanks.

Reply:
In general you want to prefer semantic and accessible HTML whenever you can. In this case it would mean having a Link.

But I think your solution is okay(ish) and I can understand your reasoning. Though the onClick should not be on an article (this is my main concern here).

And while it is true that you should not nest a button inside a button or inside a link, neither should you give onClick events to elements that aren't buttons / interactive elements. It makes your application less accessible and semantic, and in 98% of the cases, you can work with a button instead.

So in this case, I would give the onClick to the button element and style it so that it looks like you want it to look.

Reply:
Hello, as discussed I solved it by changing the article to a button and the previously included button to a simple icon. Thanks for your help.